### PR TITLE
sdk(sync-agent): move useCasablancaTimelines to sdk

### PR DESCRIPTION
**Pull request by miguel-nascimento**
Move `useCasablancaTimelines` & `use-timeline-store` to sync agent.
Updates playground to add support for redactions, threads and reactions.

close TOWNS-11921

### CODE REVIEW

File: packages/sdk/src/sync-agent/timeline/models/timelineEvents.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { Observable } from '../../../observable/observable'
-import { TimelineEvent } from './timelineEvent'
+import { type TimelineEvent } from './timeline-types'
 
 export class TimelineEvents extends Observable<TimelineEvent[]> {
```
Comment (miguel-nascimento) [inline packages/sdk/src/sync-agent/timeline/models/timelineEvents.ts:4]: I would love to see this becoming a persisted observable

File: packages/sdk/src/sync-agent/timeline/models/timeline-types.ts [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+    // | SpaceImageEvent // TODO: implement
+    // | SpaceParentEvent // TODO: understand
+    | SpaceUsernameEvent
+    | SpaceDisplayNameEvent
+    | SpaceEnsAddressEvent
+    | SpaceNftEvent
+    | RoomMessageEncryptedRefEvent
+    | UnpinEvent
+
+export enum RiverEvent {
```
Comment (miguel-nascimento) [inline packages/sdk/src/sync-agent/timeline/models/timeline-types.ts:95]: Any suggestion to name this?

File: packages/sdk/src/sync-agent/timeline/models/timeline-types.ts [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+    | RoomMemberEvent // TODO: understand
+    | RoomMessageEvent
+    // | RoomNameEvent // TODO: understand
+    | RoomPropertiesEvent // TODO: understand (can we change the name to ChannelPropertiesEvent?)
+    // | RoomTopicEvent // TODO: understand
+    | ChannelCreateEvent // NOTE: prev SpaceChild
+    // | SpaceUpdateAutojoinEvent  // TODO: implement
+    // | SpaceUpdateHideUserJoinLeavesEvent // TODO: implement
+    // | SpaceImageEvent // TODO: implement
+    // | SpaceParentEvent // TODO: understand
```
Comment (miguel-nascimento) [inline packages/sdk/src/sync-agent/timeline/models/timeline-types.ts:87]: I left a bunch of TODOs around here, is there any event that we dont need/its missing?

File: packages/sdk/src/sync-agent/timeline/models/timeline-types.ts [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+    properties: ChannelProperties
+}
+
+// TODO: membership here doenst map 1-1 to MembershipOp
+export enum Membership {
+    Join = 'join',
+    Invite = 'invite',
+    Leave = 'leave',
+    Ban = 'ban',
+    // NOTE: 'Knock' got removed, isnt used anywhere
```
Comment (miguel-nascimento) [inline packages/sdk/src/sync-agent/timeline/models/timeline-types.ts:230]: do we have plans to add it back?

File: packages/sdk/src/sync-agent/timeline/models/timeline-types.ts [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+    streamId?: string // in a case of an invitation to a channel with a streamId
+}
+
+// TODO: understand, what is Message.GM?
+// TODO: better naming maybe?
+export enum MessageType {
+    Text = 'm.text',
+    // GM = 'm.gm',
+    Image = 'm.Image',
+}
```
Comment (miguel-nascimento) [inline packages/sdk/src/sync-agent/timeline/models/timeline-types.ts:248]: What Message.GM is?

Comment (texuf) [reply packages/sdk/src/sync-agent/timeline/models/timeline-types.ts:248]: it's a protobuf "any" meant to allow extending the protocol with whatever you want to send.

File: packages/sdk/src/sync-agent/timeline/models/timeline-types.ts (hunk 4 above)
Comment (texuf) [reply packages/sdk/src/sync-agent/timeline/models/timeline-types.ts:230]: no

File: packages/sdk/src/sync-agent/timeline/models/timeline-types.ts (hunk 3 above)
Comment (texuf) [reply packages/sdk/src/sync-agent/timeline/models/timeline-types.ts:87]: lets clean this up in harmony. these are left over from building on matrix initially

File: packages/sdk/src/sync-agent/timeline/models/timeline-types.ts (hunk 2 above)
Comment (texuf) [reply packages/sdk/src/sync-agent/timeline/models/timeline-types.ts:95]: TimelineEventType?
some of these are unused. can you go through on harmony and cmd+shift+f on them and delete them if we're not using it?

Comment (miguel-nascimento) [reply packages/sdk/src/sync-agent/timeline/models/timeline-types.ts]: https://github.com/HereNotThere/harmony/pull/8264

**Comment by miguel-nascimento** [comment]
found some issues with the thread - moving back to draft

**Comment by miguel-nascimento** [comment]
the playground is missing the scrollback - ill create a ticket to add it later

File: packages/sdk/src/sync-agent/timeline/timeline.ts [hunk 6]
```diff
@@ -15,14 +38,24 @@ export class Timeline {
...
+        this.appendEvents(events, this.userId)
+    }
+
+    private reset() {
+        this.events.reset()
+        this.threads.reset()
+        this.threadsStats.reset()
+        this.reactions.reset()
+        // this.pendingReplacedEvents.reset()
+        this.replacedEvents.reset()
```
Comment (texuf) [inline packages/sdk/src/sync-agent/timeline/timeline.ts:58]: i don't think this should be commented out.

**Review by texuf — COMMENTED** [review]
can you look a these tests in harmony? 
image
image

**Review by texuf — APPROVED** [review]

**Pull request merged**